packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1067,6 +1102,60 @@ class SliverFillViewport extends SliverMultiBoxAdaptorWidget {
   }
 }
 
+class _SliverFractionalPadding extends SingleChildRenderObjectWidget {
+  const _SliverFractionalPadding({
+    this.viewportFraction = 0,
+    Widget sliver,
+  }) : assert(viewportFraction != null),
+       assert(viewportFraction >= 0),
+       assert(viewportFraction <= 0.5),
+       super(child: sliver);
+
+  final double viewportFraction;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) => _RenderSliverFractionalPadding(viewportFraction: viewportFraction);
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderSliverFractionalPadding renderObject) {
+    renderObject.viewportFraction = viewportFraction;
+  }
+}
+
+class _RenderSliverFractionalPadding extends RenderSliverEdgeInsetsPadding {
+  _RenderSliverFractionalPadding({
+    double viewportFraction = 0,
+  }) : assert(viewportFraction != null),
+       assert(viewportFraction <= 0.5),
+       assert(viewportFraction >= 0),
+       _viewportFraction = viewportFraction;
+
+  double get viewportFraction => _viewportFraction;
+  double _viewportFraction;
+  set viewportFraction(double newValue) {
+    assert(newValue != null);
+    if (_viewportFraction == newValue)
+      return;
+    _viewportFraction = newValue;
+    markNeedsLayout();
+  }
+
+  @override
+  EdgeInsets get resolvedPadding {
+    assert(constraints.axis != null);
+    final double paddingValue = constraints.viewportMainAxisExtent * viewportFraction;
+    switch (constraints.axis) {
+      case Axis.horizontal:
+        return EdgeInsets.symmetric(horizontal: paddingValue);
+      case Axis.vertical:
+        return EdgeInsets.symmetric(vertical: paddingValue);
+    }
+
+    assert(false, 'Unreachable');
+    return EdgeInsets.symmetric(vertical: paddingValue);
```

Comment:
return EdgeInsets.zero?

Reply:
The change is no longer necessary because the logic is moved to `_resolve`. But I wonder what the benefits of returning `EdgeInsets.zero` are?